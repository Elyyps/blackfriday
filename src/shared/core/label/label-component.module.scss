.label {
  background-color: var(--colour-version-website);
  font-size: 12px;
  font-weight: bold;
  color: $brand-primary;
  padding: 6px 10px;
  display: inline-flex;
  text-transform: uppercase;
  border-radius: $global-border-radius;
  font-family: $global-font-family-secondary;
  margin-bottom: 8px;
  position: relative;
}
.label__left-pointy {
  &::before {
    position: absolute;
    left: -28px;
    top: 0;
    bottom: 0;
    content: "";
    border: 14px solid transparent;
    border-right-color: rgba($brand-primary, 0.15);
  }
}

.label__right-pointy {
  &::after {
    position: absolute;
    right: -28px;
    top: 0;
    bottom: 0;
    content: "";
    border: 14px solid transparent;
    border-left-color: rgba($brand-primary, 0.15);
  }
}
