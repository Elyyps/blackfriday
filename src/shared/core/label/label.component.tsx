import * as React from "react";
import styles from "./label-component.module.scss";

export interface ILabelComponentComponentProps {
  children: JSX.Element;
  isLeftPointy?: boolean;
  isRightPointy?: boolean;
  onlyMobile?: boolean;
}

const LabelComponent = (props: ILabelComponentComponentProps) => (
  <div
    className={`${styles["label"]} ${props.isLeftPointy ? styles["label__left-pointy"] : ""} ${
      props.isRightPointy ? styles["label__right-pointy"] : ""
    } ${props.onlyMobile ? "uk-hidden@s" : ""}`}
  >
    {props.children}
  </div>
);

export { LabelComponent };
