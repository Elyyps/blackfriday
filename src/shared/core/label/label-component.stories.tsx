import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { LabelComponent } from "./label.component";

storiesOf("Label Component", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => (
    <LabelComponent>
      <span>"Black Friday 2019: NOG 10 dagen!"</span>
    </LabelComponent>
  ))
  .add("Basic implementation", () => (
    <LabelComponent isLeftPointy>
      <span>"Black Friday 2019: NOG 10 dagen!"</span>
    </LabelComponent>
  ))
  .add("Basic implementation", () => (
    <LabelComponent isRightPointy>
      <span>"Black Friday 2019: NOG 10 dagen!"</span>
    </LabelComponent>
  ))
  .add("Basic implementation", () => (
    <LabelComponent isRightPointy isLeftPointy>
      <span>"Black Friday 2019: NOG 10 dagen!"</span>
    </LabelComponent>
  ));
