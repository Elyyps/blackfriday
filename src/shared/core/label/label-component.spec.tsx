import React from "react";
import { shallow } from "enzyme";
import { LabelComponent } from "./label.component";

describe("[LabelComponent]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<LabelComponent text={"Black Friday 2019: NOG 10 dagen!"} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
