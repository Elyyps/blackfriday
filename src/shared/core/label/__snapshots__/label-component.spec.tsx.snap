// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LabelComponent] should render component without crashing 1`] = `
<div
  className="label  "
>
  Black Friday 2019: NOG 10 dagen!
</div>
`;
